### ISSUE TITLE
Appimage fails to run on Ubuntu Mate 20.04

### ISSUE THREAD

**Issue by Prosthet1cHead** [kiwix/kiwix-desktop#393]
Trying to run the kixix-desktop appimage on Ubuntu Mate 20.04 (pre-release).
```

$ ./kiwix-desktop_x86_64_2.0-rc4.appimage 
libGL error: MESA-LOADER: failed to open iris (search paths /usr/lib/x86_64-linux-gnu/dri:\$${ORIGIN}/dri:/usr/lib/dri)
libGL error: failed to load driver: iris
libGL error: MESA-LOADER: failed to open iris (search paths /usr/lib/x86_64-linux-gnu/dri:\$${ORIGIN}/dri:/usr/lib/dri)
libGL error: failed to load driver: iris
libGL error: MESA-LOADER: failed to open swrast (search paths /usr/lib/x86_64-linux-gnu/dri:\$${ORIGIN}/dri:/usr/lib/dri)
libGL error: failed to load driver: swrast
Library directory : "/home/user/.local/share/kiwix"
WebEngineContext used before QtWebEngine::initialize() or OpenGL context creation failed.
add widget
Failed to create OpenGL context for format QSurfaceFormat(version 2.0, options QFlags<QSurfaceFormat::FormatOption>(), depthBufferSize 24, redBufferSize -1, greenBufferSize -1, blueBufferSize -1, alphaBufferSize -1, stencilBufferSize 8, samples 0, swapBehavior QSurfaceFormat::SwapBehavior(DefaultSwapBehavior), swapInterval 1, profile  QSurfaceFormat::OpenGLContextProfile(NoProfile)) 
Aborted (core dumped)

$ [1:1:0100/000000.479159:ERROR:broker_posix.cc(41)] Invalid node channel message
```

The hardware is a Lenovo E550 laptop.
Any other information that would be relevant, please ask (I'm a long time linux user, but not a software developer).
Thanks & keep up the good work!

**Comment by kelson42** [comment]
@mgautierfr I confirm the problem with Ubuntu 20.04 with appimage. The nightly (future 2.0) does not work better, it fails the same ways. Like reported in #372, this does not work on openSUSE Tumbleweed 64 as well.

@Prosthet1cHead Thank you for reporting, as a workaround I would recommend to use the Flathub version in the meantime. Thank you for your report.

**Comment by Prosthet1cHead** [comment]
> @Prosthet1cHead Thank you for reporting, as a workaround I would recommend to use the Flathub version in the meantime.

Thanks, the flatpack version wants to pull in all of KDE, making the total install something like 1GB. Is that correct / necessary?
For now I have the kiwix plugin in chromium browser so that works for my immediate needs, although obviously not ideal long term. Thanks!

**Comment by kelson42** [comment]
> Thanks, the flatpack version wants to pull in all of KDE, making the total install something like 1GB. Is that correct / necessary?

Yes, this is how the Flatpack packaging works... and this is why I believe this will works.

**Comment by kelson42** [comment]
@mgautierfr @jetownfeve21 This is a PRIO1, so far I can see, it seems that Kiwix-Desktop appimage is totally broken on cutting edge GNU/Linux distributions.

**Comment by kelson42** [comment]
It seems to work fine now for me with Ubuntu 20.04 and https://download.kiwix.org/release/kiwix-desktop/kiwix-desktop_x86_64_2.0.3.appimage without VM. @Prosthet1cHead Maybe for you as well?

**Comment by kelson42** [comment]
@Prosthet1cHead Here we have a problem with the environnement, this is not Kiwix related I believe. A solution I have seen online is to force an other MESA driver:
```
MESA_LOADER_DRIVER_OVERRIDE=i965 ./kiwix-deskop
```

Does that work?

**Comment by kelson42** [comment]
No feedback, closing for the moment. @Prosthet1cHead feel free to reopen it if the bug is still there for you.

**Comment by KW4NP** [comment]
This is affecting me as well on a brand new installation of linux mint 20 Cinnamon. Perhaps it is an environment issue but it seems to be affecting multiple common distributions, so that would seem to indicate that there is a bad interaction with Kiwix and the current drivers that needs to be fixed. Is libGL out of date?

./kiwix-desktop_x86_64_2.0.4-1.appimage 

libGL error: MESA-LOADER: failed to open iris (search paths /usr/lib/x86_64-linux-gnu/dri:\$${ORIGIN}/dri:/usr/lib/dri)
libGL error: failed to load driver: iris
libGL error: MESA-LOADER: failed to open iris (search paths /usr/lib/x86_64-linux-gnu/dri:\$${ORIGIN}/dri:/usr/lib/dri)
libGL error: failed to load driver: iris
libGL error: MESA-LOADER: failed to open swrast (search paths /usr/lib/x86_64-linux-gnu/dri:\$${ORIGIN}/dri:/usr/lib/dri)
libGL error: failed to load driver: swrast
WebEngineContext used before QtWebEngine::initialize() or OpenGL context creation failed.
Library directory : "/home/timothy/.local/share/kiwix"
Compiled with Qt Version  5.9.5
Runtime Qt Version  5.9.5
add widget
Failed to create OpenGL context for format QSurfaceFormat(version 2.0, options QFlags<QSurfaceFormat::FormatOption>(), depthBufferSize 24, redBufferSize -1, greenBufferSize -1, blueBufferSize -1, alphaBufferSize -1, stencilBufferSize 8, samples 0, swapBehavior QSurfaceFormat::SwapBehavior(DefaultSwapBehavior), swapInterval 1, profile  QSurfaceFormat::OpenGLContextProfile(NoProfile)) 
Aborted (core dumped)

**Comment by ALittleBug** [comment]
I also observed same issue, anyone can help?
libGL: MESA-LOADER: failed to open /usr/lib/x86_64-linux-gnu/dri/swrast_dri.so: /lib/x86_64-linux-gnu/libdrm_amdgpu.so.1: undefined symbol: drmSyncobjTimelineSignal
libGL: MESA-LOADER: failed to open \$${ORIGIN}/dri/swrast_dri.so: \$${ORIGIN}/dri/swrast_dri.so: cannot open shared object file: No such file or directory
libGL: MESA-LOADER: failed to open /usr/lib/dri/swrast_dri.so: /usr/lib/dri/swrast_dri.so: cannot open shared object file: No such file or directory
libGL error: MESA-LOADER: failed to open swrast (search paths /usr/lib/x86_64-linux-gnu/dri:\$${ORIGIN}/dri:/usr/lib/dri)
libGL error: failed to load driver: swrast
GLFW error: 65543, GLX: Failed to create context: GLXBadFBConfig
WindowGLFW: Failed create window
terminate called after throwing an instance of 'std::exception'
  what():  std::exception
Aborted (core dumped)

**Comment by kelson42** [comment]
@mgautierfr you reopen because of the crash? Strongly suspect this is not related to the opengl.

**Comment by mgautierfr** [comment]
I've reopened it because we have something to investigate.
The crash may be related to the opengl problem or not.
And even on the opengl side, there is something wrong, it is a bit adventurous to say that the problem comes from a miss configuration of opengl on the system.

**Comment by vagabondcoder** [comment]
I'm new to Kiwix and just downloaded the appimage for 2.0.5, and got similar errors on my Ubuntu 21.04 system.
A quick google search led me to [this](https://superuser.com/questions/1377555/slic3r-appimage-fails-to-run-with-libgl-error-unable-to-load-driver-radeonsi) and I ran this:
`LD_PRELOAD=/usr/lib/x86_64-linux-gnu/libstdc++.so.6 ./kiwix-desktop_x86_64_2.0.5.appimage`
which seemed to work.  The main window launched and I was able to import a zim file and look at articles.

It seems the appimage has a older libsdc++ than my system so I need to force loading my system one.
Hope that helps...

**Comment by kelson42** [comment]
@vagabondcoder Does the latest nightly fixes the problem and works out of the box? http://download.kiwix.org/nightly/2021-09-22/kiwix-desktop_x86_64_2021-09-22.appimage

**Comment by vagabondcoder** [comment]
Your appimage 2021-09-22 gave me the same errors and crashed immediately. The `LD_PRELOAD` trick worked though.
I tried preloading libLLVM (which is required by my dri driver) and got:
```
LD_PRELOAD=/usr/lib/x86_64-linux-gnu/libLLVM-12.so ./kiwix-desktop_x86_64_2021-09-22.appimage 
./kiwix-desktop_x86_64_2021-09-22.appimage: /tmp/.mount_kiwix-bs6EJz/usr/bin/../lib/libstdc++.so.6: version `GLIBCXX_3.4.26' not found (required by /usr/lib/x86_64-linux-gnu/libLLVM-12.so)
```

So it looks like it needs `GLIBCXX_3.4.26` which seems to be packaged with gcc 9 and newer.

**Comment by Ubuntu-2020** [comment]
**kiwix-desktop_x86_64_2.0.5.appimage** and **[kiwix-desktop_x86_64_2021-12-14.appimage](http://download.kiwix.org/nightly/)** doesn't start in **Ubuntu 21.10**:

```
name@test:~/Documents$ ./kiwix-desktop_x86_64_2.0.5.appimage
libGL error: MESA-LOADER: failed to open r600: /usr/lib/dri/r600_dri.so: cannot open shared object file: No such file or directory (search paths /usr/lib/x86_64-linux-gnu/dri:\$${ORIGIN}/dri:/usr/lib/dri, suffix _dri)
libGL error: failed to load driver: r600
libGL error: MESA-LOADER: failed to open swrast: /usr/lib/dri/swrast_dri.so: cannot open shared object file: No such file or directory (search paths /usr/lib/x86_64-linux-gnu/dri:\$${ORIGIN}/dri:/usr/lib/dri, suffix _dri)
libGL error: failed to load driver: swrast
WebEngineContext used before QtWebEngine::initialize() or OpenGL context creation failed.
Compiled with Qt Version  5.9.5
Runtime Qt Version  5.9.5
add widget
Failed to create OpenGL context for format QSurfaceFormat(version 2.0, options QFlags<QSurfaceFormat::FormatOption>(), depthBufferSize 24, redBufferSize -1, greenBufferSize -1, blueBufferSize -1, alphaBufferSize -1, stencilBufferSize 8, samples 0, swapBehavior QSurfaceFormat::SwapBehavior(DefaultSwapBehavior), swapInterval 1, profile  QSurfaceFormat::OpenGLContextProfile(NoProfile)) 
Aborted (core dumped)
name@test:~/Documents$ [1:1:0100/000000.966299:ERROR:broker_posix.cc(41)] Invalid node channel message
Received signal 7 BUS_ADRERR 7ff5cdd66560
```

```
name@test:~/Documents$ ./kiwix-desktop_x86_64_2021-12-14.appimage
Warning: Ignoring XDG_SESSION_TYPE=wayland on Gnome. Use QT_QPA_PLATFORM=wayland to run on Wayland anyway.
libGL error: MESA-LOADER: failed to open r600: /usr/lib/dri/r600_dri.so: cannot open shared object file: No such file or directory (search paths /usr/lib/x86_64-linux-gnu/dri:\$${ORIGIN}/dri:/usr/lib/dri, suffix _dri)
libGL error: failed to load driver: r600
libGL error: MESA-LOADER: failed to open swrast: /usr/lib/dri/swrast_dri.so: cannot open shared object file: No such file or directory (search paths /usr/lib/x86_64-linux-gnu/dri:\$${ORIGIN}/dri:/usr/lib/dri, suffix _dri)
libGL error: failed to load driver: swrast
WebEngineContext used before QtWebEngine::initialize() or OpenGL context creation failed.
QGLXContext: Failed to create dummy context
Compiled with Qt Version  5.15.2
Runtime Qt Version  5.15.2
add widget
Failed to create OpenGL context for format QSurfaceFormat(version 2.0, options QFlags<QSurfaceFormat::FormatOption>(), depthBufferSize 24, redBufferSize -1, greenBufferSize -1, blueBufferSize -1, alphaBufferSize -1, stencilBufferSize 8, samples 0, swapBehavior QSurfaceFormat::DefaultSwapBehavior, swapInterval 1, colorSpace QSurfaceFormat::DefaultColorSpace, profile  QSurfaceFormat::NoProfile) 
Aborted (core dumped)
```

**Comment by kelson42** [comment]
@Ubuntu-2020 Seems to be a different bug, can please open a other ticket?

**Comment by Ubuntu-2020** [comment]
@kelson42 **[Ok](https://github.com/kiwix/kiwix-desktop/issues/745)**.

**Comment by kelson42** [comment]
https://docs.appimage.org/reference/best-practices.html?highlight=libstdc#libstdc-so-6

@mgautierfr with wich version do we compile kiwix-desktop against?

**Comment by EncryptedGiraffe** [comment]
I encountered this trying to run kiwix-desktop_x86_64_2.1.0-1.appimage on Debian 11 in a VM. (maybe this is the wrong issue to post in but I found the solution here)

applying this preload solution worked for me:
> I'm new to Kiwix and just downloaded the appimage for 2.0.5, and got similar errors on my Ubuntu 21.04 system. A quick google search led me to [this](https://superuser.com/questions/1377555/slic3r-appimage-fails-to-run-with-libgl-error-unable-to-load-driver-radeonsi) and I ran this: `LD_PRELOAD=/usr/lib/x86_64-linux-gnu/libstdc++.so.6 ./kiwix-desktop_x86_64_2.0.5.appimage` which seemed to work. The main window launched and I was able to import a zim file and look at articles.
> 
> It seems the appimage has a older libsdc++ than my system so I need to force loading my system one. Hope that helps...

**Comment by mgautierfr** [comment]
Fixed by kiwix/kiwix-build#520
